### PR TITLE
issue: 1265099 Add get ib_ctx object from if_name

### PR DESCRIPTION
**Pull request by liranoz12**

### CODE REVIEW

File: src/vma/dev/ib_ctx_handler_collection.cpp [hunk 1]
```diff
@@ -117,6 +117,25 @@ ib_ctx_handler* ib_ctx_handler_collection::get_ib_ctx(struct ibv_context* p_ibv_
...
 }
 
+ib_ctx_handler* ib_ctx_handler_collection::get_ib_ctx(const char *ifa_name)
+{
+	int fd;
+	char ib_path[IBV_SYSFS_PATH_MAX];
+
+	ib_context_map_t::iterator ib_ctx_iter;
+	for (ib_ctx_iter = m_ib_ctx_map.begin(); ib_ctx_iter != m_ib_ctx_map.end(); ib_ctx_iter++) {
+		snprintf(ib_path, sizeof(ib_path), "/sys/class/infiniband/%s/device/net/%s/ifindex", ib_ctx_iter->first->device->name, ifa_name);
```
Comment (igor-ivanov) [inline src/vma/dev/ib_ctx_handler_collection.cpp]: Can you add result verification for `snprintf` to pass security check.
Example: https://github.com/Mellanox/libvma/blob/master/src/vma/util/utils.h#L300
Key steps:
- char ib_path[IBV_SYSFS_PATH_MAX]={0};
- int n = -1; n=snprintf();
- if (likely((0 < n) && (n < (int)sizeof(ib_path))))

File: src/vma/util/utils.cpp [hunk 2]
```diff
@@ -460,6 +460,30 @@ int get_ifinfo_from_ip(const struct sockaddr& addr, char* ifname, uint32_t& iffl
...
 }
 
+int get_port_from_ifname(const char* ifname)
+{
+	int port_num, dev_id = -1, dev_port = -1;
+	// Depending of kernel version and OFED stack the files containing dev_id and dev_port may not exist.
+	// if file reading fails *dev_id or *dev_port may remain unmodified
+	char num_buf[24] = {0};
+	char dev_path[256] = {0};
+	sprintf(dev_path, VERBS_DEVICE_PORT_PARAM_FILE, ifname);
```
Comment (igor-ivanov) [inline src/vma/util/utils.cpp]: consider using `snprintf` in code security manner

File: src/vma/util/utils.cpp [hunk 3]
```diff
@@ -460,6 +460,30 @@ int get_ifinfo_from_ip(const struct sockaddr& addr, char* ifname, uint32_t& iffl
...
+	if (priv_safe_try_read_file(dev_path, num_buf, sizeof(num_buf)) > 0) {
+		dev_port = strtol(num_buf, NULL, 0); // base=0 means strtol() can parse hexadecimal and decimal
+		__log_dbg("dev_port file=%s dev_port str=%s dev_port val=%d", dev_path, num_buf, dev_port);
+	}
+	sprintf(dev_path, VERBS_DEVICE_ID_PARAM_FILE, ifname);
```
Comment (igor-ivanov) [inline src/vma/util/utils.cpp]: consider using `snprintf` in code security manner

File: src/vma/dev/net_device_table_mgr.cpp [hunk 4]
```diff
@@ -422,72 +398,60 @@ bool net_device_table_mgr::verify_eth_qp_creation(const char* ifname, uint8_t po
...
+		if (!priv_ibv_query_flow_tag_supported(qp, get_port_from_ifname(base_ifname))) {
+			p_ib_ctx->set_flow_tag_capability(true);
+		}
+		ndtm_logdbg("verified interface %s for flow tag capabilities : %s", ifname, p_ib_ctx->get_flow_tag_capability() ? "enabled" : "disabled");
+
+	} else {
+		ndtm_logdbg("QP creation failed on interface %s (errno=%d %m), Traffic will not be offloaded \n", ifname, errno);
+		int err = errno; //verify_raw_qp_privliges can overwrite errno so keep it before the call
+		if (validate_raw_qp_privliges() == 0) {
+			//// MLNX_OFED raw_qp_privliges file exist with bad value
```
Comment (igor-ivanov) [inline src/vma/dev/net_device_table_mgr.cpp]: replace `////` with `//`

**Comment by liranoz12** [comment]
@igor-ivanov,
Please see last commit

**Pull request merged**